espnet/nets/pytorch_backend/transducer/blocks.py

```diff
@@ -0,0 +1,277 @@
+"""Set of methods to create transformer-based block."""
```

Comment:
This is just a note.
Using this block scheme for transducer only is a bit inconsistent with other modules, but I just want to leave them as they are.
espnet2 does not have model-specific encoder/decoder architectures (https://github.com/espnet/espnet/tree/master/espnet2/asr), and we could consistently refactor this issue in the near future there.

Reply:
To be honest that feature was intended in espnet2 at first. However, 1) a lot of users (including myself) are still extensively using espnet1 and 2) I think it's a necessary to reproduce some paper results using transducer, thus it should also be available in espnet1.

Also, this PR also serve me as a draft. I wanted to finish my first stack of pending tasks before doing the porting and decide what should be done next for transducer in espnet2. I have several things I want to work on but each one will take some time: streaming, general and transducer optimizations for mobile deployment, own transducer implementation (including monornnt, modified encoder and prediction output combination, etc), new recipes, etc.